package/base-files/files/lib/upgrade/keep.d/base-files-essential

```diff
@@ -6,5 +6,6 @@
 /etc/profile
 /etc/shadow
 /etc/shells
+/etc/shinit
```

Comment:
Do we really need to backup this file?
I think if we do a sysupgrade with backup then the new file should be used which comes with the new image version.

Reply:
Since /etc/shinit is basically split from /etc/profile, I would think whatever treatment is given to /etc/profile should apply to /etc/shinit as well.

I assume /etc/profile is backed up because there may be user customizations (e.g. more aliases, environment variables, etc.); I would assume at least some of these customizations may go into /etc/shinit in the future (since it gets applied to all interactive shells, if the user is aware of these distinctions).

Reply:
I thought user and system alias should go to this directory. https://github.com/openwrt/openwrt/blob/430b66bbe8726a096b5db04dc34915ae9be1eaeb/package/base-files/Makefile#L60
And would be so sourced on load. Why should a user edit the /etc/shinit?
But that's just my opinion.

Reply:
/etc/profile (which sources /etc/profile.d/*.sh) is only loaded for login shells. /etc/shinit would be loaded for all interactive shells (login and non-login). (Aliases and functions added by /etc/profile do not exist in interactive, non-login shells.)

I believe /etc/profile.d was added to allow packages to add files to be sourced. I suppose a case could be made for a /etc/shinit.d; if there is interest I can add it.